src/main/java/log_in/drivers/LogInDatabaseAccess.java

```diff
@@ -0,0 +1,126 @@
+        ArrayList<String> emails = this.getData(4);
+        for (int i = 0; i < emails.size(); i++){
+            if (emails.get(i).equals(email)){
+                return true;
+            } else {
+                i++;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * this method will check if a password matches an email by checking if their indexes
+     * in the database are the same
+     * @param email the inputted email by the user to be checked
+     * @param pass the inputted password by the user to be checked that it matches the email
+     * @return boolean on if email and pass match and pertain to one user.
+     */
+    @Override
+    public boolean checkPasswordMatches(String email, String pass){
+        ArrayList<String> emails = this.getData(4);
+        ArrayList<String> passwords = this.getData(3);
+        int emailIndex;
+        int passIndex;
+        if (emails.contains(email)){
+            emailIndex = emails.indexOf(email);
+        } else {
+            return false;
+        }
+        if (passwords.contains(pass)){
+            passIndex = passwords.indexOf(pass);
+        }  else {
+            return false;
+        }
+
+        return emailIndex == passIndex;
+    }
+
+    @Override
+    public User getUser(boolean success, String email, String pass){
+        if (success){
+            ArrayList<String> emails = this.getData(4);
+            ArrayList<String> admins = this.getData(2);
+
+            int emailIndex = emails.indexOf(email);
+            String adminValueString = admins.get(emailIndex);
+
+            boolean isAdmin;
+            isAdmin = adminValueString.equals("true");
+
+            return new User(isAdmin, 0, email, pass);
+        } else {
+            return null;
+        }
+    }
+
+    //testing method see LogInTest for usage
+    @Override
+    public void addUser(User user){
+
+    }
+
+
+    /**
+     * This method will return a List of all the values in a csv file
+     * @param indexInfo the attribute of users in interest
+     * @return a List of a specific attribute of all users in a csv file
+     */
+    public ArrayList<String> getData(int indexInfo){
+        ArrayList<String[]> userInfo;
+        try {
+            userInfo = readLine(userPath);
+        } catch (Exception e){
+            System.out.println("Wrong Path");
+            userInfo = new ArrayList<>();
+        }
+
+        ArrayList<String> data = new ArrayList<String>();
+
+        for (String[] usersInfo: userInfo){
+            data.add(usersInfo[indexInfo]);
+        }
+
+        return data;
+    }
+
+    private ArrayList<String[]> readLine(String path) throws Exception {
+        ArrayList<String[]> list = new ArrayList<>();
+        try (Reader reader = Files.newBufferedReader(Path.of(path))) {
+            try (CSVReader csvReader = new CSVReader(reader)) {
+                String[] line;
+                while ((line = csvReader.readNext()) != null) {
```

Comment:
We may want to skip the header in the csv file (i think they have one). If not, disregard this.